File: app/src/main/java/com/solverlabs/worldcraft/util/GameTime.java
package com.solverlabs.worldcraft.util;

public class GameTime {
    private static long currentGameTime;
    private static long lastPlayedTime;
    private static long systemStartSessionTime;
    private static long timeOffset = 0;

    public static long getTime() {
        currentGameTime = (System.currentTimeMillis() - systemStartSessionTime) + lastPlayedTime + timeOffset;
        return currentGameTime;
    }

    public static void initTime(long lastPlayedTime2) {
        lastPlayedTime = lastPlayedTime2;
        systemStartSessionTime = System.currentTimeMillis();
    }

    public static void incTime(long time) {
        timeOffset += time;
    }
}

File: app/src/main/java/com/mcal/worldcraft/utils/GameTime.kt
package com.mcal.worldcraft.utils

object GameTime {
    private var currentGameTime: Long = 0
    private var lastPlayedTime: Long = 0
    private var systemStartSessionTime: Long = 0
    private var timeOffset: Long = 0

    @JvmStatic
    val time: Long
        get() {
            currentGameTime =
                System.currentTimeMillis() - systemStartSessionTime + lastPlayedTime + timeOffset
            return currentGameTime
        }

    @JvmStatic
    fun initTime(lastPlayedTime: Long) {
        this.lastPlayedTime = lastPlayedTime
        systemStartSessionTime = System.currentTimeMillis()
    }

    @JvmStatic
    fun incTime(time: Long) {
        timeOffset += time
    }
}